### ISSUE TITLE
readwrite

### ISSUE THREAD

**Issue by jarrodmillman** [networkx/networkx#4311]
We should probably have our read/write functionality all in one place:
- review ``write_dot`` and ``read_dot`` in ``drawing/nx_agraph.py`` and ``drawing/nx_pydot.py``
- review ``algorithms/bipartite/edgelist.py`` and ``readwrite/edgelist.py``

**Comment by dschult** [comment]
Good point! --     But it also makes sense to put all the dot stuff in one place. 

Maybe the dot/pydot/pygraphviz should all be in read/write instead of drawing?
At a fundamental level, drawing via pygraphviz **is** writing in a specific format. 

OTOH... why would you write a dot file except to draw the network?

Hmmmm.....  I don't like splitting `write_dot` from the pygraphviz interfacing code, but someone looking for all our writable formats might not find it in `drawing`.  Could we keep the code together but have the docs for write_doc listed with read/write?

**Comment by rossbar** [comment]
We should probably revisit this discussion in light of the proposed pydot deprecation. If `pydot` is removed from the equation, that leaves the `bipartite` loading functionality. Assuming the mirrored functionality is sufficiently different from its non-bipartite version, I think leaving maintaining it in the separate `bipartite` namespace makes sense. Of course, we should review to be sure this is actually the case!

**Comment by rossbar** [comment]
Yeah it gets pretty tricky... my initial attempt at categorization would be that "visualization" deals with viewing graphs live/interactively, whereas "readwrite" deals with loading/saving/transporting graphs.

Even this distinction breaks down in some cases though: for example, some visualization modalities use an intermediary text representation for the graph like LaTeX or PyDot. Then, in order to actually *visualize* the graph, you need some external backend (a latex engine or graphviz, respectively) to convert the intermediate format into an actual image.

All this is to say - I don't think there's a clear-cut way to organize things, nor would it be worthwhile to get *too* hung up on where things live. I'd just as soon approve just about any proposal in order to avoid a potential bikeshedding trap :)

**Comment by rossbar** [comment]
This one's been stale for a while so I will go ahead and close it. I think we can improve the situation on the documentation side by ensuring e.g. pydot is visible from multiple places - see #8204 for instance.
